agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2265,6 +2487,154 @@ bool backhaul_manager::handle_1905_higher_layer_data_message(ieee1905_1::CmduMes
     return send_cmdu_to_bus(cmdu_tx, src_mac, bridge_info.mac);
 }
 
+bool backhaul_manager::handle_1905_link_metric_query(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                                     const std::string &src_mac)
+{
+    const auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received LINK_METRIC_QUERY_MESSAGE, mid=" << std::hex << int(mid);
+
+    /**
+     * The IEEE 1905.1 standard says about the Link Metric Query TLV and the neighbor type octet
+     * that "If the value is 0, then the EUI48 field is not present; if the value is 1, then the
+     * EUI-48 field shall be present."
+     *
+     * However, optional fields are not currently supported by TLVF.
+     *
+     * As a workaround, instead of defining a tlvLinkMetricQuery TLV with an optional field, we
+     * have defined two different TLVs: tlvLinkMetricQuery with the optional field and
+     * tlvLinkMetricQueryAllNeighbors without it. Application must check which of both TLVs has
+     * been received inside the message.
+     */
+    std::shared_ptr<ieee1905_1::tlvLinkMetricQueryAllNeighbors> tlvLinkMetricQueryAllNeighbors;
```

Comment:
Perhaps we can handle each option in separate functions? (nitpick)

Reply:
Most of the code is common to both options. Separated functions would contain lots of duplicated code so I prefer to leave it as it is now.